Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/FacepunchMultiplayerMenu/FacepunchMultiplayerMenu.cs

```diff
@@ -0,0 +1,210 @@
+﻿#if FACEPUNCH_STEAMWORKS
+using BeardedManStudios.Forge.Networking;
+using BeardedManStudios.Forge.Networking.Unity;
+using BeardedManStudios.Forge.Logging;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+using Steamworks;
+using System.Threading.Tasks;
+
+
+public class FacepunchMultiplayerMenu : MonoBehaviour
+{
+	public const int MAXIMUM_SERVER_SLOTS = 64;
+
+	public bool dontChangeSceneOnConnect = false;
+	public GameObject networkManager = null;
+	public GameObject[] toggledButtons;
+	public Steamworks.Data.Lobby lobbyToJoin = default;
+	public bool useMainThreadManagerForRPCs = true;
+	public bool useInlineChat = false;
+
+	private NetworkManager mgr = null;
+	private NetWorker server;
+	private Steamworks.Data.Lobby lobby;
+	private List<Button> _uiButtons = new List<Button>();
+
+	private void Start()
+	{
+		for (int i = 0; i < toggledButtons.Length; ++i)
+		{
+			Button btn = toggledButtons[i].GetComponent<Button>();
+			if (btn != null)
+				_uiButtons.Add(btn);
+		}
+
+		if (useMainThreadManagerForRPCs)
+			Rpc.MainThreadRunner = MainThreadManager.Instance;
+	}
+
+	/// <summary>
+	/// Connects to the currently selected Steam Lobby
+	/// Lobby selection is made within FacepuncJoinMenu.cs
+	/// </summary>
+	public void Connect()
+	{
+		if (lobbyToJoin.Id > 0)
+		{
+			ConnectToLobbyAsync();
+		}
+
+		// TODO:  Raise a dialog to inform of the failed connection attempt
+	}
+
+	/// <summary>
+	/// Connect to selected steam lobby asynchronously
+	/// Void method calls the async task
```

Comment:
The second line has no added value as the code clearly shows what is going on :wink: